services/product-service/serverless.ts

```diff
@@ -44,13 +48,51 @@ const serverlessConfiguration: AWS = {
               "arn:aws:dynamodb:us-east-1:*:table/stocks",
             ],
           },
+          {
+            Effect: "Allow",
+            Resource: {
+              Ref: "createProductTopic",
+            },
+            Action: ["sns:*"],
+          },
         ],
       },
     },
   },
   // import the function via paths
-  functions: { getProductsById, getProductsList, createProduct },
+  functions: {
+    getProductsById,
+    getProductsList,
+    createProduct,
+    catalogBatchProcess,
+  },
   package: { individually: true },
+  resources: {
+    Resources: {
+      catalogItemsQueue: {
+        Type: "AWS::SQS::Queue",
+        Properties: {
+          QueueName: "catalogItemsQueue",
+        },
+      },
+      createProductTopic: {
+        Type: "AWS::SNS::Topic",
+        Properties: {
+          TopicName: "createProductTopic",
+        },
+      },
+      createProductSubscription: {
+        Type: "AWS::SNS::Subscription",
+        Properties: {
+          Protocol: "email",
+          Endpoint: "orsolya.matisz07@gmail.com",
```

Comment:
its a public repo , that's why not recommended to write personal info like email here.
Alternatively you could put in .env file and use useDotenv: true in serverless.ts 
with Endpoint: '${env:EMAIL}',